R/parameters.R

```diff
@@ -1,3 +1,48 @@
+##' Create a linear time-varying beta for input into 
+##' \code{create_parameters()}
+##' 
+##' @title Generate beta
+##' 
+##' @param beta_start Value of beta before intervention
+##' 
+##' @param start_date Start date for simualtions
+##' 
+##' @param reduction_start Start of reduction in beta
+##' 
+##' @param beta_reduction Factor by which beta is reduced
+##'   default is 0.238, as estimated by Jarvis et al
+##'   beta_reduced = beta - beta * (1 - beta_reduction)
+##' 
+##' @param reduction_period Time, in days, over which the
+##' reduction in beta is achieved. Default 10 days
+##' 
+##' @export
+generate_beta <- function(beta_start, 
+                          start_date = "2020-02-02",
+                          reduction_start = "2020-03-16",
+                          beta_reduction = 0.238, 
+                          reduction_period = 10) {
+  if (as.Date(start_date) > as.Date(reduction_start)) {
+    stop("Start date must be earlier than intervention date")
+  }
+  if (beta_reduction < 0 || beta_reduction > 1) {
+    stop("beta must decrease, and cannot be reduced below 0")
+  }
+  if (reduction_period > 100) {
+    message("Reduction period over 100 days - is this correct?")
```

Comment:
I like this message, but we might reconsider in 3 months  📆

Reply:
Is that right? Would there potentially be a smooth reduction over such a long time?